worker/pkg/benthos/transformers/generate_state.go

```diff
@@ -10,11 +11,21 @@ import (
 // +neosyncTransformerBuilder:generate:generateState
 
 func init() {
-	spec := bloblang.NewPluginSpec().Description("Randomly selects a US state and returns the two-character state code.")
-
+	spec := bloblang.NewPluginSpec().Param(bloblang.NewBoolParam("state_code").Optional()).Description("Randomly selects a US state and either returns the two character state code or the full state name.")
```

Comment:
Instead of using `Optional()` you can replace it with `Default(false)` to better describe the default behavior.

Reply:
Also, if you check out some other generator like `generate_ssn`, you can see how the spec is formatted and how it includes a top level description as well as a description for each parameter. This provides well formed generated documentation for this generator.